share/functions/path_helper.fish

```diff
@@ -0,0 +1,22 @@
+# Reimplement the macOS /usr/libexec/path_helper executable for fish;
+# see __fish_construct_path in config.fish and
+# https://opensource.apple.com/source/shell_cmds/shell_cmds-203/path_helper/path_helper.c.auto.html .
+
+function __quote -d "single-quote arguments, escaping characters as necessary"
```

Comment:
I couldn't find a built-in way to quote a string. Thought `string escape` would do it, but it does something different.

Reply:
`string escape` _will_ do what you want - escape a string so that it'll pass through `source` unharmed. It just doesn't always use single-quotes. Instead it falls back to using `\\` for more complicated things.

However, I'm not quite sure why this `path_helper` function exists at all - you're not using it anywhere as far as I can see, and since it would run in the fish process, there's no need to output the set statements - instead, you could just directly run them. Which would also remove the need for quoting.

Reply:
It's just a driver for debugging the path_helper logic without having to start shells and examine environment variables. I can remove it if needed.

Reply:
Yes, please! This looks like something to invoke, and it really shouldn't be.

Reply:
Done! Please take another look.